dali/pipeline/operators/reader/loader/sequence_loader.h

```diff
@@ -93,27 +93,28 @@ class SequenceLoader : public Loader<CPUBackend, TensorSequence> {
         stride_(spec.GetArgument<int32_t>("stride")),
         streams_(filesystem::GatherExtractedStreams(file_root_)),
         sequences_(detail::GenerateSequences(streams_, sequence_length_, step_, stride_)),
-        total_size_(sequences_.size()),
-        current_sequence_(0) {
+        total_size_(sequences_.size()) {
     DALI_ENFORCE(sequence_length_ > 0, "Sequence length must be positive");
     DALI_ENFORCE(step_ > 0, "Step must be positive");
     DALI_ENFORCE(stride_ > 0, "Stride must be positive");
     mmap_reserver = FileStream::FileStreamMappinReserver(
         static_cast<unsigned int>(initial_buffer_fill_) * sequence_length_);
     copy_read_data_ = !mmap_reserver.CanShareMappedData();
-    if (shuffle_) {
-      // seeded with hardcoded value to get
-      // the same sequence on every shard
-      std::mt19937 g(524287);
-      std::shuffle(sequences_.begin(), sequences_.end(), g);
-    }
+    // seeded with hardcoded value to get
```

Comment:
you dropped the `if (shuffle_)`. Was it intentional?

Reply:
Yes. It was wrong from the very beginning. We should always do initial shuffling to mix data, so GPU0 won't get only 0 class data for a few initial batches.

Reply:
Grr, it breaks test. Video loader is no better. Reverting to original code. It need to be fixed separately.

Reply:
My bad, it looks we are doing the same for other reader and it was meant to be like that.